docs/src/main/asciidoc/hibernate-orm-panache.adoc

```diff
@@ -1,326 +1,752 @@
+    }
+    public void setStatus(Status status) {
+        this.status = status;
+    }
+}
+----
+. Define your repository and add instance methods. The following example defines the `PersonRepository` :
++
+[source,java]
+----
+@ApplicationScoped
+public class PersonRepository implements PanacheRepository<Person> {
+
+   public Person findByName(String name){
+       return find("name", name).firstResult();
+   }
+
+   public List<Person> findAlive(){
+       return list("status", Status.Alive);
+   }
+
+   public void deleteStefs(){
+       delete("name", "Stef");
+  }
+}
+----
+
++
+. Add the following code to the REST endpoint where you want to use entity operations so that the operations that are defined in `PanacheEntityBase` are available:
++
+[source,java]
+----
+@Inject
+PersonRepository personRepository;
+
+@GET
+public long count(){
+    return personRepository.count();
+}
+----
+
+=== Pananche common operations
+This section lists common operations that you can use with your entity or repository. The active record pattern examples use a `Person` entity created from the `PanacheEntity` class. The repository pattern examples use a regular JPA `Person` entity and the `PersonRepository` created from the
+`PanacheRepository` class.
+
+* Create a record (active record pattern and repository pattern):
++
 [source,java]
 ----
-// creating a person
 Person person = new Person();
 person.name = "Stef";
 person.birth = LocalDate.of(1910, Month.FEBRUARY, 1);
 person.status = Status.Alive;
+----
 
-// persist it
-person.persist();
+* Persist an entity:
+** Active record pattern:
++
+[source,java]
+----
+person.persist(person);
+----
 
-// note that once persisted, you don't need to explicitly save your entity: all
-// modifications are automatically persisted on transaction commit.
+** Repository pattern:
++
+[source,java]
+----
+personRepository.persist(person);
+----
++
+NOTE: After an entity is persisted, you do not need to explicitly save your entity. All changes are automatically persisted on transaction commit.
 
-// check if it's persistent
-if(person.isPersistent()){
-    // delete it
-    person.delete();
-}
+* Verify that an entity is persisted:
+** Active record pattern:
++
+[source,java]
+----
+if(person.isPersistent(person))
+----
+** Repository pattern:
++
+[source,java]
+----
+if(personRepository.isPersistent(person))
+----
 
-// getting a list of all Person entities
-List<Person> allPersons = Person.listAll();
+* Delete an entity:
+** Active record pattern:
++
+[source,java]
+----
+person.delete(person);
+----
 
-// finding a specific person by ID
-person = Person.findById(personId);
+** Repository pattern:
++
+[source,java]
+----
+personRepository.delete(person);
+----
 
-// finding a specific person by ID via an Optional
-Optional<Person> optional = Person.findByIdOptional(personId);
+* Retrieve a list of all `Person` entities:
+** Active record pattern:
++
+[source,java]
+----
+List<Person> allPersons = person.listAll();
```

Comment:
```suggestion
List<Person> allPersons = Person.listAll();
```

Generall the case difference between `person` and `Person` is meaningful, so you should review all those samples to keep them correct: `person` is the instance, and `person.foo()` is an instance method call (a call on that particular object), while `Person` is the class and `Person.foo()` is a static method (a call that does not target a particular instance).